lib/extractor/layer.ts

```diff
@@ -53,6 +55,70 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
+  ociArchiveFilesystemPath: string,
+  extractActions: ExtractAction[],
+): Promise<{
+  layers: ExtractedLayers[];
+  manifest: OciArchiveManifest;
+}> {
+  return new Promise((resolve, reject) => {
+    const tarExtractor: Extract = extract();
+
+    const layers: Record<string, ExtractedLayers> = {};
+    const manifests: OciArchiveManifest[] = [];
+    let imageIndex;
+
+    tarExtractor.on("entry", async (header, stream, next) => {
+      if (header.type === "file") {
```

Comment:
not critical by any mean, just personal preference: instead of having 
```
if (header.type === "file") {
lots of lots of code
}
```
I prefer to have
```
if (header.type !== "file") {
    return;
}
rest of code
```

Reply:
💯

Reply:
Actually I can't really do that! 😬 `return` leaves the block so we don't get any more events and when I tried to use `next()` I was getting some VERY unexpected results!

Reply:
Ah yes, instead of `return` you have to consume the stream! Like it was done here: https://github.com/snyk/snyk-docker-plugin/blob/50264f304f1fd4d2362db11813ab1617ade4bfe9/lib/extractor/layer.ts#L40-L41

I'm OK with leaving it as is anyway